src/wampy.js

```diff
@@ -559,81 +559,73 @@ class Wampy {
      * @private
      */
     _packPPTPayload (payload, options) {
-        const payloadItems = [];
-        let err = false, argsList, argsDict;
-
-        if (Array.isArray(payload)) {
-            argsList = payload;
-        } else if (this._isPlainObject(payload)) {
-            // It's a wampy unified form of payload passing
-            if (payload.argsList || payload.argsDict) {
-                const isArgsListInvalid = payload.argsList && !Array.isArray(payload.argsList);
-                const isArgsDictInvalid = payload.argsDict && !this._isPlainObject(payload.argsDict);
-
-                if (isArgsListInvalid || isArgsDictInvalid) {
-                    const invalidParameter = isArgsListInvalid ? payload.argsList : payload.argsDict;
-                    err = true;
-                    this._fillOpStatusByError(new Errors.InvalidParamError(invalidParameter));
-                    return { err, payloadItems };
-                }
+        const isArgsListInvalid = payload?.argsList && !Array.isArray(payload.argsList);
+        const isArgsDictInvalid = payload?.argsDict && !this._isPlainObject(payload.argsDict);
 
-                argsList = payload.argsList;
-                argsDict = payload.argsDict;
-            } else {
-                argsDict = payload;
-            }
-        } else {    // assume it's a single value
-            argsList = [payload];
+        if (isArgsListInvalid || isArgsDictInvalid) {
+            const invalidParameter = isArgsListInvalid ? payload.argsList : payload.argsDict;
+            this._fillOpStatusByError(new Errors.InvalidParamError(invalidParameter));
+            return { err: true, payloadItems: [] };
         }
 
-        // Check and handle Payload PassThru Mode
-        // @see https://wamp-proto.org/wamp_latest_ietf.html#name-payload-passthru-mode
-        if (options.ppt_scheme) {
-            const pptPayload = { args: argsList, kwargs: argsDict };
-            let binPayload;
+        const isPayloadAnObject = this._isPlainObject(payload);
+        const { argsList, argsDict } = payload || {};
+        let args, kwargs;
 
-            if (options.ppt_serializer && options.ppt_serializer !== 'native') {
-                const pptSerializer = this._options.payloadSerializers[options.ppt_serializer];
+        if (isPayloadAnObject && !argsList && !argsDict) {
+            kwargs = payload;
+        } else if (isPayloadAnObject) {
+            args = argsList;
+            kwargs = argsDict;
+        } else {
+            args = Array.isArray(payload) ? payload : [payload];
+        }
```

Comment:
Mb change it to this, so we can keep a simple one-level if-else block without inner checks which we can not avoid.

```suggestion
        } else if (Array.isArray(payload)) {
            args = payload;
        } else {  // assume it's a single value
            args = [payload];
        }
```